### ISSUE TITLE
Add debugging fields when we log out to users that an ACL is denied

### ISSUE THREAD

**Issue by mkcp** [hashicorp/consul#7010]
Initial research done by @adams-sarah on this ticket, with impl. additions from @mkeeler.

Per [support requests](https://app.asana.com/0/211891323705885/1129595442717157) and user feedback #4706, we're aware of users having problems debugging the behavior of ACL-restricted agents. They are unable to associate events denied by ACL policies to the clients initiating the requests.

In order to resolve this. We'd like to add the `AccessorID` and `Description` of the token being denied to the log output we log out to users. One current implementaiton we have is use a `structs.ACLToken`'s `SecretID` to request the rest of its fields, via the [`acl/token/self` HTTP endpoint](https://www.consul.io/api/acl/tokens.html#read-self-token). We then filter the response from the API down to the keys and values for `AccessorID` and `Description` and return a hashmap to the location we're logging out the debug-data. The latency involved in this request isn't great performance wise. But, for the sake of delivering something workable for users, we're operating under the assumption that the perf hit is the worth the cost. (Discussion here is welcome!)

One challenge with the `acl/token/self/` approach is that we require access to the `SecretID` in order to resolve the remaining fields in the API. By searching for the phrase "blocked by ACLs" we've identified 7 locations where we inform operators that an ACL token was denied. (`agent/agent.go`, `agent/consul/internal_endpoint.go`, `agent/local/state.go`) Both `agent.go` and `internal_endpoint.go` have the SecretID in scope when we log out the message, so we may simply add a func call to grab the debug context. However, `agent/local/state.go` does not, and provides the highest user value with the debug-data added. There is still some more research to be done on how we should bring the `SecretID` into scope for the `Check` and `Service` operations in `local`, but shoutout to @adams-sarah for helping me explore that part of the codebase, and guidance on how to write tests to run that section of code w/ `ACLs` enabled on the `agent`. 

In a future ticket, (maybe related to recent HClog additions?) we may wish to make these log lines compatible with log-level configuration. I would consider this out of scope for the current issue, but a valuable addition to scope out in another issue.

**Comment by mkcp** [comment]
Edited mentions of the current `WARN` prefix to change it to `DEBUG` logging -- it's a simple string change, so it's easy to put in scope for this issue. Will make future changes to add log-level configuration easier as well by clearly stating its debug intent.

**Comment by mkeeler** [comment]
@mkcp One minor point. Instead of issuing the RPC to retrieve the tokens directly, it would be beneficial to go through the `ACLResolver` as it will cache things and prevent subsequent permission denied errors in quick succession from issuing more RPCs (dependent on the configured `token_ttl`).

**Comment by mkcp** [comment]
While reviewing #7028 , spotted a cluster of ACL-denied type logs in `agent/consul/acl.go`. 

Findable by searching for `"due to ACLs"`
https://github.com/hashicorp/consul/blob/master/agent/consul/acl.go#L1185

Not sure if it's in scope for this ticket yet.

**Comment by mkcp** [comment]
@mkeeler There's a whole bunch of places now that we have access to an `authz` and are logging out that some operation is blocked/denied due to ACLs. Not sure if it's in scope for this addition, but would it make sense to add the `accessorID` of the Authorizer's token to the `acl.Authorizer` itself?

**Comment by mkcp** [comment]
I believe this is resolved by #7117, closing
